pandas/io/excel/_xlrd.py

```diff
@@ -95,9 +105,30 @@ def _parse_cell(cell_contents, cell_typ):
                     cell_contents = val
             return cell_contents
 
-        data = []
+        data: List[List[Scalar]] = []
 
+        if nrows is not None:
```

Comment:
This block of code looks repeated in the various excel files. Could we turn it into a shared function?